server/repository/src/migrations/v1_08_00/sync_file_reference.rs

```diff
@@ -0,0 +1,19 @@
+use crate::{migrations::*, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+            CREATE TABLE sync_file_reference (
+                id TEXT NOT NULL PRIMARY KEY,
+                table_name TEXT NOT NULL, -- Associated Table
+                record_id TEXT NOT NULL, -- Associated record id
+                file_name TEXT NOT NULL,
+                created_datetime TIMESTAMP NOT NULL, -- No modified datetime, as we don't allow updates it would break sync
+                deleted_datetime TIMESTAMP
```

Comment:
Do we want to capture other things in here?
Maybe a category for the file in case the same record has more than one type of file associated with it?
Mime type might be handy?
File description could be good too, but depends if we're going to create UI to manage these?

Reply:
some sort of type would be good. For the CCEI example, we need to distinguish an image and a document.
`mime_type` would be a good option - allows the consumer to make decisions and possibly categorise